menas/ui/service/OozieService.js

```diff
@@ -13,19 +13,64 @@
  * limitations under the License.
  */
 
+jQuery.sap.require("sap.m.MessageToast");
 jQuery.sap.require("sap.m.MessageBox");
 
 var OozieService = new function () {
 
-  const model = sap.ui.getCore().getModel();
+  const model = () => {
+    return sap.ui.getCore().getModel();
+  }
   const eventBus = sap.ui.getCore().getEventBus();
 
   this.getCoordinatorStatus = function () {
-    const coordinatorId = model.getProperty("/currentDataset/schedule/activeInstance/coordinatorId")
+    const coordinatorId = model().getProperty("/currentDataset/schedule/activeInstance/coordinatorId")
     if(coordinatorId) {
-      Functions.ajax(`api/oozie/coordinatorStatus/${coordinatorId}`, "GET", {}, function (oData) {
-        model.setProperty("/currentDataset/schedule/activeInstance/status", oData)
-      }, function () {
+      RestClient.get(`api/oozie/coordinatorStatus/${coordinatorId}`).then((oData) => {
```

Comment:
I'd like the `RestClient` to be abstracted away, as much as possible as well as the URIs so that they can be changed in a single location.
Let's extract these into an `OozieRestDAO`:
- `RestClient.get(api/oozie/coordinatorStatus/${coordinatorId})` => `OozieRestDAO.getStatus(coordinatorId)`
- `RestClient.post("api/oozie/runNow", oSchedule)` => `OozieRestDAO.postSchedule(oSchedule)`
... etc